lib/classes/Variables.js

```diff
@@ -54,6 +54,7 @@ class Variables {
     this.optRefSyntax = RegExp(/^opt:/g);
     this.selfRefSyntax = RegExp(/^self:/g);
     this.stringRefSyntax = RegExp(/(?:('|").*?\1)/g);
+    this.boolRefSyntax = RegExp(/(?:(true|false))/g);
```

Comment:
If I see correctly with this regex we allow strings that contain `true` or `false` strings, and not necessary are equal to them.

In result `foofalsebar` will be translated to `false`.

I take that our stringRefSyntax is also susceptible to same issue

Reply:
@medikoo you're right, I updated the regex to match word exactly matching and added a test case to ensure your exemple is covered. I don't see how stringRefSyntax is affected as well. Unlike bool, string syntax uses ' or " delimiters.

Reply:
With string I believe problem is that `foo"string"bar` will match (I think, I haven't tested), but this is not necessarily concern for this PR